src/addons/transitions/ReactCSSTransitionGroupChild.js

```diff
@@ -48,8 +48,8 @@ var ReactCSSTransitionGroupChild = React.createClass({
 
   transition: function(animationType, finishCallback) {
     var node = React.findDOMNode(this);
-    var className = this.props.name + '-' + animationType;
-    var activeClassName = className + '-active';
+    var className = this.props.name[animationType] || this.props.name + '-' + animationType;
+    var activeClassName = this.props.name[animationType + 'Active'] || className + '-active';
```

Comment:
Ah, sorry! I missed this comment. Let's skip the test for now. Just make sure `appear` works and this should be good to go.